application/src/main/java/org/togetherjava/tjbot/commands/tags/TagCommand.java

```diff
@@ -62,4 +69,20 @@ public void onSlashCommand(SlashCommandInteractionEvent event) {
         }
         message.queue();
     }
+
+    @Override
+    public void onAutoComplete(CommandAutoCompleteInteractionEvent event) {
+        AutoCompleteQuery focusedOption = event.getFocusedOption();
+
+        if (focusedOption.getName().equals(ID_OPTION)) {
+            Collection<Command.Choice> choices = StringDistances
+                .autocompleteSuggestions(focusedOption.getValue(), tagSystem.getAllIds(), 0.5)
+                .stream()
+                .map(id -> new Command.Choice(id, id))
+                .limit(OptionData.MAX_CHOICES)
+                .toList();
+
+            event.replyChoices(choices).queue();
```

Comment:
`tagSystem.getAllIds()` can take a bit since it has to check the database. not sure how autocomplete works with long methods, but in regular commands u have to use `deferReply()` and then respond via the interactionhook. maybe u have to do sth like that here as well?

Reply:
Auto completion should be fast, I don't think there's deferReply to begin with, but even if there was, having to wait 3s for auto completion is way too long if you ask me.

Reply:
`CommandAutoCompleteInteractionEvent` have no method called `deferReply()` also i don't think `replyChoices()` timesout